util/argo/audit_logger.go

```diff
@@ -157,10 +179,45 @@ func (l *AuditLogger) LogAppProjEvent(proj *v1alpha1.AppProject, info EventInfo,
 	l.logEvent(objectMeta, v1alpha1.AppProjectSchemaGroupVersionKind, info, message, nil)
 }
 
-func NewAuditLogger(ns string, kIf kubernetes.Interface, component string) *AuditLogger {
+func NewAuditLogger(ns string, kIf kubernetes.Interface, component string, enableK8sEvent []string) *AuditLogger {
 	return &AuditLogger{
-		ns:        ns,
-		kIf:       kIf,
-		component: component,
+		ns:             ns,
+		kIf:            kIf,
+		component:      component,
+		enableEventLog: setK8sEventList(enableK8sEvent),
+	}
+}
+
+func setK8sEventList(enableK8sEvent []string) map[string]bool {
+	enableK8sEventList := make(map[string]bool)
+
+	sort.Slice(enableK8sEvent, func(i, j int) bool {
+		return enableK8sEvent[i] > enableK8sEvent[j]
+	})
+
+	for _, event := range enableK8sEvent {
+		if event == "all" {
+			enableK8sEventList = map[string]bool{
+				EventReasonStatusRefreshed:    true,
+				EventReasonResourceCreated:    true,
+				EventReasonResourceUpdated:    true,
+				EventReasonResourceDeleted:    true,
+				EventReasonResourceActionRan:  true,
+				EventReasonOperationStarted:   true,
+				EventReasonOperationCompleted: true,
+			}
+			return enableK8sEventList
+		} else if event == "none" {
+			enableK8sEventList = map[string]bool{}
+			return enableK8sEventList
+		}
+
+		enableK8sEventList[event] = true
 	}
+
+	return enableK8sEventList
+}
+
+func TestEnableEvent() []string {
+	return []string{"all"}
 }
```

Comment:
This function was added because of its dependency on other unit tests.
Renaming the function to DefaultEnableEventList looks good.
I think this will solve the hard-coded part with []string{“all”} and the part I wrote with TestEnableEvent.

Reply:
That should work as well. We can then use the function `DefaultEnableEventList` wherever default value needs to be set.